application/controllers/RegisterController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+// use Google\Service\CloudSearch\Resource\Query;
+require_once __DIR__ . '/../models/QueryModel.php';
+
+class RegisterController {
+  private $db;
+
+  function __construct() {
+    $this->db = new QueryModel();
+  }
+
+  function invoke() {
+    // If POST request.
+    if(isset($_POST['task'])) {
+      if($_POST['task'] === 'submit-user-data') {
+        $this->getUserData();
+      }
+      else if($_POST['task'] === 'submit-otp') {
+        $this->validateOtp();
+      }
+    }
+    else {
+      require_once __DIR__ . '/../views/register_view.php';
+    }
+  }
+
+  public function getUserData() {
+    session_start();
+    $_SESSION['email'] = $_POST['email'];
+    $_SESSION['fname'] = $_POST['fname'];
+    $_SESSION['lname'] = $_POST['lname'];
+    $_SESSION['password'] = $_POST['password'];
```

Comment:
Instead of storing them in separate variables
Can we maybe do something like this to reduce cluster

```
    $_SESSION['userData']['email'] = $_POST['email'];
    $_SESSION['userData']['fname'] = $_POST['fname'];
    $_SESSION['userData']['lname'] = $_POST['lname'];
    $_SESSION['userData']['password'] = $_POST['password'];
```